test/suite/hook_tiered.py

```diff
@@ -416,23 +422,53 @@ def setup_hooks(self):
 
         self.wiredtiger['wiredtiger_open'] = (wthooks.HOOK_ARGS, wiredtiger_open_tiered)
 
+# Strip matching single or double quotes
+def strip_quotes(s):
+    if len(s) >= 2:
+        if (s[0] == '\"' or s[0] == '\'') and s[0] == s[-1]:
+            s = s[1:-1]
+    return s
+
+def config_split(config):
+    pos = config.index('=')
+    if pos >= 0:
+        left = config[:pos]
+        right = config[pos+1:]
+    else:
+        left = config
+        right = ''
+    return [left, strip_quotes(right)]
+
 # Override some platform APIs for this hook.
 class TieredPlatformAPI(wthooks.WiredTigerHookPlatformAPI):
     def __init__(self, arg=None):
         self.tier_share_percent = 0
         self.tier_cache_percent = 0
         self.tier_storage_source = 'dir_store'
         params = []
+
+        # We want to split args something like arg.split(','), except that we need
+        # to sometimes allow commas as part of the individual parameters, which we
+        # allow via quoting.  For example, a developer can run:
+        #  run.py --hook \
+        #    'tiered=tier_storage_source=dir_store,tier_storage_source_config="force_delay=5,delay_ms=10"'
```

Comment:
I agree with the sentiment, trying to figure out how to get clarity without a lot of rewrite.  In using parens, we'd need to recognize them (parens currently are no more special than any other character, so `tiered=(tier_storage_source=dir_store)` would currently be parsed as key: `(tiered_storage_source` is assigned value `dir_store)`.  For using quotes, we currently allow quotes in a specific position (around the contents of the value), rather than something more general like what /bin/sh does with quoting.

In this example we have really three levels of '=', which I agree is confusing.  I think a reasonable compromise is to use parens rather than double quotes.  (To get double quotes to work at multiple levels we'd have use and recognize backslashes or something equivalent.  Then you have multiple backslashes which is hard to decipher).  With parens recognized at the different levels we could have something more legible.  Also trying to make this be cognitively compatible with WT configuration's use of parens, which is really more like a list.
```
run.py --hook \
     'tiered=(tier_storage_source=dir_store,tier_storage_source_config=(force_delay=5,delay_ms=10))'
```
The () perform like quotations, and like WT we'd only recognize them immediately after a '='.  So at each level of parsing we'd have for example _A=a,B=b,C=c_ and _a_ is not allowed to have '=' or ',' unless _a_ is surrounded by ().  If that's the case _A=(x=1,y=2,z=(foo=(bar=blam)))_ for example, we strip off the outside parens and that new string is used, and can be similarly parsed by the lower level.

I think that conforms best with WT use of () and sets us up for whatever complexity we need.
What do you think @pmacko86?

Reply:
Sounds good to me! Thank you.